training/src/main/java/wolox/training/security/SecurityConfig.java

```diff
@@ -0,0 +1,54 @@
+package wolox.training.security;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.builders.WebSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import wolox.training.security.CustomAuthenticationProvider;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+
+    @Autowired
+    private CustomAuthenticationProvider authProvider;
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+        auth.authenticationProvider(authProvider);
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception{
+
+        http
+            .csrf().disable()
+            .exceptionHandling()
+            .and()
+            .authorizeRequests()
+            .anyRequest().authenticated()
+            .and()
+            .httpBasic();
+    }
+
+    @Override
+    public void configure(WebSecurity web) {
+        web.ignoring().antMatchers(HttpMethod.POST, "/api/users/");
+        web.ignoring().antMatchers(HttpMethod.POST, "/api/books/");
```

Comment:
This one should be authenticated, we don't want anybody creating books ;)

Reply:
> De todos los recursos que disponés, vamos a dejar sin autenticación los de creación de Book y de creación de User.
:thinking:

Reply:
Never mind, me sa am stupid 😆